nservicebus/hosting/critical-errors.md

```diff
@@ -19,49 +21,49 @@ partial: default
- * The code should be wrapped in a `try...finally` clause. In the `try` block perform any custom operations; in the `finally` block call the method that exits the process.
- * The custom operations should include flushing any in-memory state and cached data, if normally it is persisted at a certain interval or during graceful shutdown. For example, flush appenders when using buffering or asynchronous appenders for [NLog](https://nlog-project.org/documentation/v4.3.0/html/M_NLog_LogManager_Shutdown.htm) or [log4net](https://logging.apache.org/log4net/log4net-1.2.11/release/sdk/log4net.LogManager.Shutdown.html) state by calling `LogManager.Shutdown();`.
+* To exit the process use the [Environment.FailFast](https://msdn.microsoft.com/en-us/library/dd289240.aspx) method. In case the environment has threads running that should be completed before shutdown (e.g. non transactional operations), the [Environment.Exit](https://msdn.microsoft.com/en-us/library/system.environment.exit.aspx) method can also be used.
+* The code should be wrapped in a `try...finally` clause. In the `try` block perform any custom operations; in the `finally` block call the method that exits the process.
+* The custom operations should include flushing any in-memory state and cached data, if normally it is persisted at a certain interval or during graceful shutdown. For example, flush appenders when using buffering or asynchronous appenders for [NLog](https://nlog-project.org/documentation/v4.3.0/html/M_NLog_LogManager_Shutdown.htm) or [log4net](https://logging.apache.org/log4net/log4net-1.2.11/release/sdk/log4net.LogManager.Shutdown.html) state by calling `LogManager.Shutdown();`.
 
 Whenever possible rely on the environment hosting the endpoint process to automatically restart it:
 
- * IIS: The IIS host will automatically spawn a new instance.
- * Windows Service: The OS can restart the service after 1 minute if [Windows Service Recovery](/nservicebus/hosting/windows-service.md#installation-setting-the-restart-recovery-options) is enabled.
+* IIS: The IIS host will automatically spawn a new instance.
+* Windows Service: The OS can restart the service after 1 minute if [Windows Service Recovery](/nservicebus/hosting/windows-service.md#installation-setting-the-restart-recovery-options) is enabled.
 
-WARNING: It is important to consider the effect these defaults will have on other things hosted in the same process. For example if co-hosting NServiceBus with a web-service or website.
+## Terminating the process
 
+In general the best strategy is to terminate the process when a critical error occurs and rely on the process hosting environment to restart the process as a recovery mechanism. Reason this is the best strategy is that it is unknown when a critical error occurs if the issues is recoverable. By assuming it is not and terminating the process results in the more resilient way on dealing with critical errors.
```

Comment:
Here are just some grammar changes to what you wrote:
```suggestion
In general, the best strategy is to terminate the process when a critical error occurs and rely on the process hosting environment to restart the process as a recovery mechanism. The reason this is the best strategy is that it is unknown when a critical error occurs if the issue is recoverable. By assuming it is not and terminating the process results in a more resilient way of dealing with critical errors.
```

I question the "best". We know it isn't the best for everyone, but I think we can present terminating the process as a good choice. I attempted to reword it:

```suggestion
It is often unknown when a critical error occurs if the issue is recoverable. A sound strategy is to terminate the process when a critical error occurs and rely on the process hosting environment to restart the process as a recovery mechanism, resulting in a resilient way to deal with critical errors.```